pkg/runtime/framework/plugins/torch/torch_test.go

```diff
@@ -0,0 +1,1014 @@
+			info: runtime.NewInfo(
+				runtime.WithMLPolicy(
+					utiltesting.MakeMLPolicyWrapper().
+						WithNumNodes(2). // This value should be overridden by the value in trainJob
+						TorchPolicy("auto", nil).
+						Obj(),
+				),
+				runtime.WithLabels(map[string]string{
+					"app": "pytorch-training",
+					"env": "production",
+				}),
+			),
+			wantInfo: &runtime.Info{
+				Labels: map[string]string{
+					"app": "pytorch-training",
+					"env": "production",
+				},
+				Annotations: make(map[string]string),
+				RuntimePolicy: runtime.RuntimePolicy{
+					MLPolicy: utiltesting.MakeMLPolicyWrapper().
+						WithNumNodes(2).
+						TorchPolicy("auto", nil).
+						Obj(),
+				},
+				Trainer: runtime.Trainer{
+					NumNodes:       ptr.To[int32](4),
+					NumProcPerNode: "",
+					Env: []corev1ac.EnvVarApplyConfiguration{
+						{
+							Name:  ptr.To(constants.TorchEnvNumNodes),
+							Value: ptr.To("4"),
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvNumProcPerNode),
+							Value: ptr.To("auto"),
+						},
+						{
+							Name: ptr.To(constants.TorchEnvNodeRank),
+							ValueFrom: &corev1ac.EnvVarSourceApplyConfiguration{
+								FieldRef: &corev1ac.ObjectFieldSelectorApplyConfiguration{
+									FieldPath: ptr.To(constants.JobCompletionIndexFieldPath),
+								},
+							},
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvMasterAddr),
+							Value: ptr.To("gpu-job-trainer-node-0-0.gpu-job"),
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvMasterPort),
+							Value: ptr.To(fmt.Sprintf("%d", constants.ContainerTrainerPort)),
+						},
+					},
+					ContainerPort: &corev1ac.ContainerPortApplyConfiguration{
+						ContainerPort: ptr.To[int32](constants.ContainerTrainerPort),
+					},
+				},
+				Scheduler: &runtime.Scheduler{TotalRequests: map[string]runtime.TotalResourceRequest{}},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			var cancel func()
+			ctx, cancel = context.WithCancel(ctx)
+			t.Cleanup(cancel)
+			cliBuilder := utiltesting.NewClientBuilder()
+			p, err := New(ctx, cliBuilder.Build(), nil)
+			if err != nil {
+				t.Fatalf("Failed to initialize Torch plugin: %v", err)
+			}
+
+			// Test EnforceMLPolicy
+			err = p.(framework.EnforceMLPolicyPlugin).EnforceMLPolicy(tc.info, tc.trainJob)
+			if diff := cmp.Diff(tc.wantMLPolicyError, err, cmpopts.EquateErrors()); len(diff) != 0 {
+				t.Errorf("Unexpected error from EnforceMLPolicy (-want,+got):\n%s", diff)
+			}
+
+			// Validate the entire info object
+			if tc.wantInfo != nil {
+				if diff := cmp.Diff(tc.wantInfo, tc.info,
+					cmpopts.SortSlices(func(a, b string) bool { return a < b }),
+					cmpopts.SortMaps(func(a, b string) bool { return a < b }),
+				); len(diff) != 0 {
+					t.Errorf("Unexpected RuntimeInfo (-want,+got):\n%s", diff)
+				}
+			} else if tc.info != nil {
+				t.Errorf("Expected info to be nil, but got: %v", tc.info)
+			}
```

Comment:
```suggestion
			if diff := cmp.Diff(tc.wantInfo, tc.info,
				cmpopts.SortSlices(func(a, b string) bool { return a < b }),
				cmpopts.SortMaps(func(a, b string) bool { return a < b }),
			); len(diff) != 0 {
				t.Errorf("Unexpected RuntimeInfo (-want,+got):\n%s", diff)
			}
```

Why can't we do this one?